colour/algebra/interpolation.py

```diff
@@ -1217,36 +1189,26 @@ def _evaluate(self, x: NDArrayFloat) -> NDArrayFloat:
 
         r = self._yp
 
-        a0p = r[i]
-        a1p = (2 * r[i - 2] - 16 * r[i - 1] + 16 * r[i + 1] - 2 * r[i + 2]) / 24
-        a2p = (-r[i - 2] + 16 * r[i - 1] - 30 * r[i] + 16 * r[i + 1] - r[i + 2]) / 24
-        a3p = (
-            -9 * r[i - 2]
-            + 39 * r[i - 1]
-            - 70 * r[i]
-            + 66 * r[i + 1]
-            - 33 * r[i + 2]
-            + 7 * r[i + 3]
-        ) / 24
-        a4p = (
-            13 * r[i - 2]
-            - 64 * r[i - 1]
-            + 126 * r[i]
-            - 124 * r[i + 1]
-            + 61 * r[i + 2]
-            - 12 * r[i + 3]
-        ) / 24
-        a5p = (
-            -5 * r[i - 2]
-            + 25 * r[i - 1]
-            - 50 * r[i]
-            + 50 * r[i + 1]
-            - 25 * r[i + 2]
-            + 5 * r[i + 3]
-        ) / 24
-
-        y = a0p + a1p * X + a2p * X**2 + a3p * X**3 + a4p * X**4 + a5p * X**5
+        r_s = np.asarray((r[i - 2], r[i - 1], r[i], r[i + 1], r[i + 2], r[i + 3]))
+        w_s = np.asarray(
+            (
+                (2, -16, 0, 16, -2, 0),
+                (-1, 16, -30, 16, -1, 0),
+                (-9, 39, -70, 66, -33, 7),
+                (13, -64, 126, -124, 61, -12),
+                (-5, 25, -50, 50, -25, 5),
+            )
+        )
+        a = np.dot(w_s, r_s) / 24
+
+        # Fancy vector code here... use underlying numpy structures to accelerate
+        # parts of the linear algebra.
 
+        y = r[i] + np.einsum(
+            "ab,ab->b", a.reshape(5, -1), X ** np.arange(1, 6).reshape(-1, 1)
+        )
```

Comment:
With the desire to move toward data-array API compliance and because Einstein Sum is not planned to be supported, we should try to rewrite this using lower level APIs: https://data-apis.org/array-api/latest/extensions/linear_algebra_functions.html

Reply:
I removed all usages of `np.einsum` in `develop`.